database/mongodb/transformData.js

```diff
@@ -0,0 +1,38 @@
+const fs = require('fs');
+const csv = require('csv-parser');
+const path = require('path');
+const client = require('./index.js');
+const { Product, Style } = require('./models.js');
+
+const db = mongoose.connection;
+
+
+async function transformData() {
+  try {
+    db.on('error', console.error.bind(console, 'MongoDB Connection error:'));
+    db.once('open', async () => {
+      const csvFiles = ['product.csv', 'features.csv', 'related.csv'];
+
+      for (const csvFile of csvFiles) {
+        const filePath = path.resolve(`database/data/${csvFile}`);
+
+        fs.createReadStream(filePath)
+          .pipe(csv())
+          .on('data', async (data) => {
+            console.log(data);
+            await Product.findOneAndUpdate({ productId: data.id }, data, { upsert: true });
```

Comment:
I suppose you're trying to merge all product data from these three csv's here into one mongo model (Products)? You will need to make sure the objects returned for each row have the appropriate attributes to match your Products schema. Also double check that the list fields (features, related) are being appended to and not overwritten with the upsert option.